### DISCUSSION TITLE
RSA keygeneration takes long

### DISCUSSION THREAD

**Discussion by androidseb25** [krzyzanowskim/CryptoSwift#1063] [Q&A]
After updating to 1.8.4 keygeneration for RSA with a keysize 2048 takes up to 10s for generating!! I used the swift package manager and implement it for my iOS apps. 

I tried a smaller keysize like 1024 and it takes 4s.

After updating to the last version (1.8.4) it slows down, other libraries with 1.8.3 runs fast 🤔

**Comment by krzyzanowskim** [comment]
there was no changes to rsa (or any major change) that could affect that diff https://github.com/krzyzanowskim/CryptoSwift/compare/1.8.3...1.8.4 Normally slow means swift debug build. looks like xcode uncached something?

**Comment by androidseb25** [comment]
i think it's nothing with caching because i cleared derrived data and installed XCode new.
also a new project for testing didn't changed anything :/
